### DISCUSSION TITLE
A blog comment system built with a frontpage

### DISCUSSION THREAD

**Discussion by syui** [likeandscribe/frontpage#186] [Show and tell]
The frontpage is absolutely amazing !

I've never seen such a simple and clear atproto project. The code and readme are particularly nice and to the point. The frontpage seems to run perfectly.

So I decided to try using it for my blog comment system.

The blog is built with [hugo](https://github.com/gohugoio/hugo). For example, you can use it by putting `$rkey` in `post.md`. 

([zola](https://github.com/getzola/zola) might be a better choice now...)

- https://syui.ai/blog/post/2024/10/26/frontpage/

```html
+++
title = "The frontpage is absolutely amazing"
date = "2024-10-26"
[params]
  comment = "3l7lprmoka22r"
+++
```

```html
<!-- layouts/_default/single.html --> 
{{ if .Param "comment" }}
	<iframe class="frontpage" src="https://example.com/post/{{ .Page.Author }}/{{ .Params.comment }}" frameBorder="0" SRC=""></iframe>
{{ end }}
```

If you know of a better way to do this other than using an iframe, please let me know.

[google translate]

**Comment by tom-sherman** [comment]
That's super cool!

> If you know of a better way to do this other than using an iframe, please let me know.

Probably nothing that's super easy right now, but in the future we'll be releasing a public API (like Bluesky's) that will allow you to fetch the comment thread for a post.

**Reply by syui** [reply]
That's amazing!

**Comment by avinassh** [comment]
This is a great idea! I use Hugo too and I will probably try this!

**Comment by WillCorrigan** [comment]
Thank you for your kind words ♥️
